packages/store/src/Types.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+// TODO: add remaining types
+enum SchemaType {
+  None, // The first `None` value ends the schema
+  Uint8,
+  Uint16,
+  Uint32,
+  Uint128,
+  Uint256,
+  Bytes4,
+  Uint32Array,
+  Bytes24Array,
+  String,
+  Address,
+  AddressArray
+}
+
+/**
+ * Get the length of the data for the given schema type
+ * (Because Solidity doesn't support constant arrays, we need to use a function)
+ * TODO: add more types and make it more efficient (avoid linear search)
```

Comment:
avoiding both linear and binary searches is probably a good idea, something like https://github.com/dk1a/mud/blob/c4cae23ac0284d6d475b683ad145daaf748e910e/packages/solecs/v2/SchemaType.sol#L225